packages/network-console-shared/src/file_io/postman/v2.1/collection-adapter.ts

```diff
@@ -0,0 +1,192 @@
+        this._dirty = true;
+    }
+
+    get authorization() {
+        return new AuthorizationAdapter(this._current, () => { this._dirty = true; });
+    }
+
+    set authorization(value: INetConsoleAuthorization) {
+        delete this._current.auth;
+        const adapter = this.authorization;
+        migrateAuthorization(adapter, value);
+        this._dirty = true;
+    }
+
+    get childEntryIds() {
+        return Array.from(this._keyToIndex.keys());
+    }
+
+    getEntryById(id: string): ICollectionEntryAdapter | null {
+        const index = this._keyToIndex.getByKey(id);
+        if (typeof index !== 'number') {
+            return null;
+        }
+
+        const entry = this._current.item[index] as Postman21Entry;
+        if ('request' in entry) {
+            return new RequestAdapter(this.format, this, id, entry, () => { this._dirty = true; });
+        }
+        else if ('item' in entry && Array.isArray(entry.item)) {
+            return new ContainerAdapter(this.format, this, id, entry, () => { this._dirty = true; });
+        }
+
+        throw new Error(`Parser error: Child with id "${id}" was not a valid request or collection folder.`);
+    }
+
+    async appendContainerEntry(name: string): Promise<ICollectionContainerAdapter> {
+        const folder: Postman21Entry = {
+            name,
+            item: [],
+        };
+        const index = this._current.item.length;
+        this._current.item.push(folder);
+        const id = `${this.id}/${this._nextKey++}`;
+        this._keyToIndex.set(id, index);
+        this._dirty = true;
+
+        return new ContainerAdapter(this.format, this, id, folder, () => { this._dirty = true; });
+    }
+
+    async appendItemEntry(request: INetConsoleRequest): Promise<ICollectionItemAdapter> {
+        const item: Postman21Entry = mapNCToPostman(request);
+        const index = this._current.item.length;
+        this._current.item.push(item);
+        const id = `${this.id}/${this._nextKey++}`;
+        this._keyToIndex.set(id, index);
+        this._dirty = true;
+
+        return new RequestAdapter(this.format, this, id, item, () => { this._dirty = true; });
+    }
+
+    async deleteEntry(id: string) {
+        const index = this._keyToIndex.getByKey(id);
+        if (typeof index !== 'number') {
+            throw new RangeError(`ID "${id}" not found in this container.`);
+        }
+        this._keyToIndex.deleteByKey(id);
+        this._current.item.splice(index, 1);
+        this._dirty = true;
+
+        // Move all values up by one
+        const valuesList = Array.from(this._keyToIndex.values()).filter(v => v > index);
+        valuesList.sort();
+        // This will never error because it's traversed from least to greatest.
+        for (const oldIndex of valuesList) {
+            const key = this._keyToIndex.getByValue(oldIndex)!;
+            this._keyToIndex.deleteByKey(key);
+            this._keyToIndex.set(key, oldIndex - 1);
+        }

```

Comment:
We can't give items an ID (we had in a previous iteration), because IDs don't actually belong to other file formats that we want to support, and expecting human-editable files to maintain IDs consistency is problematic.  I'm amenable to modifying this, or giving BidiMap some specialization that abstracts away this particular operation, but giving IDs directly to each request is a non-starter.

Reply:
I've created a specialization of `BidiMap` called `IdIndexMap` and authored a test suite for it. That results in removing all of the other instances of this particular kind of pattern.